File: src/main/java/wraith/fabricaeexnihilo/json/ingredient/EntityTypeIngredientJson.java
package wraith.fabricaeexnihilo.json.ingredient;

import com.google.gson.JsonDeserializationContext;
import com.google.gson.JsonElement;
import com.google.gson.JsonParseException;
import com.google.gson.JsonSerializationContext;
import wraith.fabricaeexnihilo.api.crafting.EntityTypeIngredient;
import wraith.fabricaeexnihilo.json.BaseJson;

import java.lang.reflect.Type;

public final class EntityTypeIngredientJson extends BaseJson<EntityTypeIngredient> {

    private EntityTypeIngredientJson() {}

    public static final EntityTypeIngredientJson INSTANCE = new EntityTypeIngredientJson();

    @Override
    public EntityTypeIngredient deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
        return EntityTypeIngredient.fromJson(json);
    }

    @Override
    public JsonElement serialize(EntityTypeIngredient src, Type typeOfSrc, JsonSerializationContext context) {
        return src.toJson();
    }

}

File: src/main/kotlin/exnihilofabrico/json/EntityTypeIngredientJson.kt
package exnihilofabrico.json

import com.google.gson.*
import com.google.gson.reflect.TypeToken
import exnihilofabrico.api.crafting.EntityTypeIngredient
import java.lang.reflect.Type

object EntityTypeIngredientJson: JsonDeserializer<EntityTypeIngredient>,
    JsonSerializer<EntityTypeIngredient> {
    val TYPE_TOKEN: Type = object: TypeToken<EntityTypeIngredient>(){}.type
    @Throws(JsonParseException::class)
    override fun deserialize(json: JsonElement, typeOfT: Type, context: JsonDeserializationContext) =
        EntityTypeIngredient.fromJson(json, context)

    @Throws(JsonParseException::class)
    override fun serialize(src: EntityTypeIngredient, typeOfSrc: Type, context: JsonSerializationContext) = src.toJson(context)
}